zio-schema/shared/src/main/scala/zio/schema/DynamicValue.scala

```diff
@@ -0,0 +1,1066 @@
+          //This should never happen unless someone manually builds an Enum and doesn't include all cases
+          case _ => DynamicValue.NoneValue
+        }
+
+      case Schema.EnumN(cases) =>
+        cases
+          .find(_.deconstruct(value).isDefined) match {
+          case Some(c) =>
+            DynamicValue.Enumeration(
+              c.id -> fromSchemaAndValue(c.codec.asInstanceOf[Schema[Any]], c.unsafeDeconstruct(value))
+            )
+          case None => DynamicValue.NoneValue
+        }
+
+      case Schema.Fail(message) => DynamicValue.Error(message)
+
+      case Schema.Sequence(schema, _, toChunk) =>
+        DynamicValue.Sequence(toChunk(value).map(fromSchemaAndValue(schema, _)))
+
+      case Schema.EitherSchema(left, right) =>
+        value match {
+          case Left(a)  => DynamicValue.LeftValue(fromSchemaAndValue(left, a))
+          case Right(b) => DynamicValue.RightValue(fromSchemaAndValue(right, b))
+        }
+
+      case Schema.Tuple(schemaA, schemaB) =>
+        val (a, b) = value
+        DynamicValue.Tuple(fromSchemaAndValue(schemaA, a), fromSchemaAndValue(schemaB, b))
+
+      case Schema.Optional(schema) =>
+        value match {
+          case Some(value) => DynamicValue.SomeValue(fromSchemaAndValue(schema, value))
+          case None        => DynamicValue.NoneValue
+        }
+
+      case Schema.Transform(schema, _, g) =>
+        g(value) match {
+          case Left(message) => DynamicValue.Transform(DynamicValue.Error(message))
+          case Right(a)      => DynamicValue.Transform(fromSchemaAndValue(schema, a))
+        }
+
+      case Schema.CaseObject(instance) => DynamicValue.Singleton(instance)
+
+      case Schema.CaseClass1(_, f, _, ext) =>
+        DynamicValue.Record(ListMap(f.label -> fromSchemaAndValue(f.schema, ext(value))))
+
+      case Schema.CaseClass2(_, f1, f2, _, ext1, ext2) =>
+        DynamicValue.Record(
+          ListMap(
+            f1.label -> fromSchemaAndValue(f1.schema, ext1(value)),
+            f2.label -> fromSchemaAndValue(f2.schema, ext2(value))
+          )
+        )
+      case Schema.CaseClass3(_, f1, f2, f3, _, ext1, ext2, ext3) =>
+        DynamicValue.Record(
+          ListMap(
+            f1.label -> fromSchemaAndValue(f1.schema, ext1(value)),
+            f2.label -> fromSchemaAndValue(f2.schema, ext2(value)),
+            f3.label -> fromSchemaAndValue(f3.schema, ext3(value))
+          )
+        )
+      case Schema.CaseClass4(_, f1, f2, f3, f4, _, ext1, ext2, ext3, ext4) =>
+        DynamicValue.Record(
+          ListMap(
+            f1.label -> fromSchemaAndValue(f1.schema, ext1(value)),
+            f2.label -> fromSchemaAndValue(f2.schema, ext2(value)),
+            f3.label -> fromSchemaAndValue(f3.schema, ext3(value)),
+            f4.label -> fromSchemaAndValue(f4.schema, ext4(value))
+          )
+        )
+      case Schema.CaseClass5(_, f1, f2, f3, f4, f5, _, ext1, ext2, ext3, ext4, ext5) =>
```

Comment:
Since performance is not incredibly important for `DynamicValue`, I think it's probably ok to just base these on the conversion to the generic record case. It can delete a lot of code.

Reply:
yeah, that's what I wanted to do originally but we need the `extractFieldN`s in order to pull values our of the instance of `Z`